### ISSUE TITLE
Feature Request: Support Comprehensive Logging Functionality for AWS WAF

### ISSUE THREAD

**Issue by vladgolubev** [terraform-providers/terraform-provider-aws#5760]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS released a feature for AWS WAF which allows wiring up Kinesis Firehose stream to route all the logs about incoming requests and matched rules. 

https://aws.amazon.com/about-aws/whats-new/2018/08/aws-waf-launches-new-comprehensive-logging-functionality/

<details>
<summary>Screenshot</summary>

![image](https://user-images.githubusercontent.com/3817380/44977864-e56b9b80-af71-11e8-884c-a9771247720e.png)

</details>

### New or Affected Resource(s)

* aws_waf_web_acl
* aws_wafregional_web_acl

### Potential Terraform Configuration

```hcl
resource "aws_waf_web_acl" "waf_acl" {
  depends_on = [
    "aws_waf_ipset.ipset",
    "aws_waf_rule.wafrule",
  ]

  name        = "tfWebACL"
  metric_name = "tfWebACL"

  logging {
    firehose_stream_arn = "${aws_kinesis_firehose_delivery_stream.test.arn}"

    redacted_fields {
      http_method  = true
      query_string = true
      uri          = true

      headers = [
        "Accept",
        "Content-Type",
      ]
    }
  }

  default_action {
    type = "ALLOW"
  }

  rules {
    action {
      type = "BLOCK"
    }

    priority = 1
    rule_id  = "${aws_waf_rule.wafrule.id}"
    type     = "REGULAR"
  }
}

```

**Comment by guipaiva** [comment]
+1

**Comment by anilkasu** [comment]
Any update on this enhancement, when can we expect to release this? Any chances including this feature in v0.12?

**Comment by jgrevich** [comment]
+1

**Comment by bflad** [comment]
I provided an initial pull request review of #6059 which adds logging support for the `aws_waf_web_acl` resource and submitted a pull request (#7480) which does the same for the `aws_wafregional_web_acl` resource.

> Any update on this enhancement, when can we expect to release this?

The `aws_waf_web_acl` resource pull request will be reviewed again when the community contributor has completed the feedback items or the maintainers will finish and release it in two weeks if there is no response. The `aws_wafregional_web_acl` resource pull request needs another maintainer review and can likely be released next week.

> Any chances including this feature in v0.12?

@anilkasu please note since Terraform 0.10, providers are no longer distributed as part of Terraform Core and have their own release cadence. Here are the [Terraform AWS Provider CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md) and the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) for reference.

**Comment by bflad** [comment]
Support for a new `logging_configuration` configuration block has been added to both the `aws_waf_web_acl` and `aws_wafregional_web_acl` resources which will release in version 1.59.0 of the Terraform AWS Provider, likely in the next day or two. 👍

**Comment by bflad** [comment]
This has been released in [version 1.59.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1590-february-14-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
